src/main/java/step3/main.java

```diff
@@ -0,0 +1,40 @@
+package step3;
+
+import java.util.Map;
+
+import step3.domain.LottoOrder;
+import step3.domain.WinningNumber;
+import step3.domain.Rank;
+import step3.service.LottoStatisticsService;
+import step3.service.LottoValidateService;
+import step3.view.InputView;
+import step3.view.OutputView;
+
+public class main {
+    public static void main(String[] args) {
+        // 초기 Input 및 Output class 선언
+        InputView inputView = new InputView();
+        OutputView outputView = new OutputView();
+
+        // 로또 구매 금액 입력 및 로또 발행
+        int price = inputView.init();
+        LottoOrder lottoOrder = new LottoOrder(price);
+        outputView.printNumbers(lottoOrder);
+
+        // 로또 당첨 번호 입력 및 우승자 찾기
+        String winnerNumber = inputView.inputWinner();
+        int bonusNumber = inputView.inputBonusNumber();
+        WinningNumber winningNumber = new WinningNumber(winnerNumber, bonusNumber);
+
+        // 로또 검증 서비스 생성 및 검증
+        LottoValidateService validateService = new LottoValidateService(lottoOrder.lottos(), winningNumber);
+        validateService.validateAll();
+        Map<Rank, Integer> result = validateService.result();
```

Comment:
```suggestion
        Map<Rank, Integer> result = validateService.result();
```

```java
public class LottoValidateService {
...
    public Map<Rank, Integer> result() {
        return winner.matchResultMap2(bought);
    }
}

public class WinningNumber {
...
    public Map<Rank, Integer> matchResultMap2(Lottos bought) {
        Map<Lotto, Rank> matchResultMap = bought.matchResultMap2(lottoNumbers, bonusNumber);
        return matchResultMap.entrySet()
            .stream()
            .collect(
                Collectors.groupingBy(
                    Map.Entry::getValue,
                    Collectors.summingInt(e -> 1)
                )
            );
}
```

`validateService.validateAll();`가 수행되고 `validateService.result();`수행되어야하는 순차적인 코드 호출은 유지보수시 힘든 요소가 될수있습니다. 이런 로직의 본질이 무엇인지 고민해보시면 좋을것 같습니다.

값을 변경,맵핑하기보다는 상태를 가진 객체에게 메시지를 보낼수 없을지 고민해보아요~
> 반영을 위한 코멘트 아닙니다. 이런 관점도 있다는것을 보여드릴려는것입니다~